src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs

```diff
@@ -394,6 +394,15 @@ protected Task<ImmutableArray<Document>> FindDocumentsWithForEachStatementsAsync
             }, cancellationToken);
         }
 
+        protected Task<ImmutableArray<Document>> FindDocumentsWithDeconstructionAsync(Project project, IImmutableSet<Document> documents, CancellationToken cancellationToken)
```

Comment:
:bulb: Consider refactoring these two similar methods to use a helper that takes a `Func<WhateverTheIndexTypeIs, bool>`. At this point I wouldn't move the helper to a different class or anything, but it does help show the common pattern for this type of document lookup.